partner_credit_limit/model/sale.py

```diff
@@ -15,7 +15,7 @@ def check_limit(self):
                 return True
             allowed_sale = self.env['res.partner'].with_context(
                 {'new_amount': so.amount_total,
-                 'new_currency': so.company_id.currency_id.id}).browse(
+                 'new_currency': so.currency_id}).browse(
```

Comment:
not necessary browse you can use

``` python
allowed_sale = po.partner_id.with_context(
 {'new_amount': so.amount_total,
'new_currency': so.currency_id}).allowed_sale
```

could you check if there are another cases?